FinSearchUnified/Tests/Components/ProductStream/RepositoryTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace FinSearchUnified\Tests\Components\ProductStream;
+
+use FinSearchUnified\Components\ProductStream\Repository;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Components\Test\Plugin\TestCase;
+
+class RepositoryTest extends TestCase
+{
+    /**
+     * @return array
+     */
+    public function shopSearchSwitchProvider()
+    {
+        return [
+            'Uses the original implementation' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => false,
+                'isCategoryPage' => true,
+                'prepareCriteria' => true
+            ],
+            'Uses the original implementation for backend' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => true,
+                'isCategoryPage' => false,
+                'prepareCriteria' => false
+            ],
+            'Uses the custom implementation' => [
+                'ActivateFindologic' => true,
+                'ShopKey' => 'ABCD0815',
+                'ActivateFindologicForCategoryPages' => false,
+                'findologicDI' => false,
+                'isSearchPage' => true,
+                'isCategoryPage' => false,
+                'prepareCriteria' => false,
+            ]
+        ];
+    }
+
+    /**
+     * @dataProvider shopSearchSwitchProvider
+     *
+     * @param bool $prepareCriteria
+     */
+    public function testIfPrepareCriteriaIsCalled($prepareCriteria)
+    {
+        $mockedRepository = $this->getMockBuilder('\Shopware\Components\ProductStream\Repository')
+            ->setMethods(['prepareCriteria'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        if ($prepareCriteria) {
+            $mockedRepository->expects($this->atLeastOnce())
+                ->method('prepareCriteria');
+        } else {
+            $mockedRepository->expects($this->never())
+                ->method('prepareCriteria');
+        }
+
+        $repository = new Repository($mockedRepository);
+        $repository->prepareCriteria(new Criteria(), 1);
```

Comment:
Why can the second parameter be  1? Shouldn't it cause an error since it is supposed to be of type `ShopContextInterface`?

Reply:
No the second parameter is the `productStreamId` not the context. Unless I am mistaken, please let me know if I am creating the correct `Repository` object.

Reply:
Nope, you are correct. I just checked the wrong class.